diff --git a/target/linux/airoha/image/an7581.mk b/target/linux/airoha/image/an7581.mk
index d9a72ef..a1e1c90 100644
--- a/target/linux/airoha/image/an7581.mk
+++ b/target/linux/airoha/image/an7581.mk
@@ -99,3 +99,27 @@ define Device/nokia_xg-040g-md
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
 endef
 TARGET_DEVICES += nokia_xg-040g-md
+
+define Device/bell_xg-040g-md
+  $(call Device/FitImageLzma)
+  DEVICE_VENDOR := Nokia
+  DEVICE_MODEL := Bell XG-040G-MD
+  DEVICE_DTS := an7581-bell_xg-040g-md
+  DEVICE_DTS_CONFIG := config@1
+  SOC := an7581
+  KERNEL_LOADADDR := 0x80088000
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  KERNEL_SIZE := 8192k
+  IMAGE_SIZE := 261120k
+  KERNEL_IN_UBI := 1
+  UBINIZE_OPTS := -s 2048
+  IMAGES := factory.bin sysupgrade.bin
+  IMAGE/factory.bin := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-ubi
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+  DEVICE_PACKAGES := kmod-phy-airoha-en8811h kmod-i2c-an7581 \
+	kmod-leds-gpio kmod-gpio-button-hotplug uboot-envtools ubi-utils \
+	kmod-usb-storage-uas kmod-usb-net-cdc-ether kmod-fs-vfat \
+	kmod-fs-exfat kmod-fs-ext4 kmod-fs-ntfs3 blkid lsblk
+endef
+TARGET_DEVICES += bell_xg-040g-md
